### PR TITLE
Add a check for changing oml which are higher than changer got.

### PR DESCRIPTION
**Pull request by reiterl**
Add a test.

### CODE REVIEW

**Comment by reiterl** [comment]
I prevent the change of the omls, if the changer of the oml is lower oml of the user.
So a superadmin cannot be downgraded by an admin.

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py [hunk 1]
```diff
@@ -343,10 +343,25 @@ def check_group_D(
...
+        """Check Group E organization_management_level: OML level necessary.
+        Also checks if user has higher oml."""
         if fields and permstore.user_oml < OrganizationManagementLevel.CAN_MANAGE_USERS:
             raise MissingPermission(OrganizationManagementLevel.CAN_MANAGE_USERS)
+        user = self.datastore.get(
+            FullQualifiedId(Collection("user"), instance["id"]),
+            ["organization_management_level"],
+        )
+        if (
+            fields
```
Comment (jsangmeister) [inline openslides_backend/action/actions/user/create_update_permissions_mixin.py:358]: The restriction should not only apply for changing the OML of another user, but for editing him altogether (that's what I tried to say with https://github.com/OpenSlides/openslides-backend/issues/883#issuecomment-890905647). So if I have OML can_manage_users and X has OML can_manage_organization, then I cannot call `user.update` for him, indipendent on the action data.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/action/actions/user/create_update_permissions_mixin.py:358]: So it should be a more general check? Okay.

Comment (reiterl) [reply openslides_backend/action/actions/user/create_update_permissions_mixin.py]: Worked on this.
The permission to change a user with higher oml is fully restricted.

Comment (reiterl) [reply openslides_backend/action/actions/user/create_update_permissions_mixin.py]: I also fixed the test problems.

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py [hunk 2]
```diff
@@ -346,28 +347,31 @@ def check_group_D(
...
 
     def check_group_F(self, fields: List[str]) -> None:
         """Group F: OML SUPERADMIN necessary, which is checked before"""
         if fields:
             raise MissingPermission(OrganizationManagementLevel.SUPERADMIN)
 
+    def _check_for_higher_OML(
+        self, permstore: PermissionVarStore, instance: Dict[str, Any]
+    ) -> None:
+        if "id" in instance:
```
Comment (jsangmeister) [inline openslides_backend/action/actions/user/create_update_permissions_mixin.py:362]: You can remove this check, id must be in instance

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py [hunk 3]
```diff
@@ -346,28 +347,31 @@ def check_group_D(
...
+            user = self.datastore.get(
+                FullQualifiedId(Collection("user"), instance["id"]),
+                ["organization_management_level"],
+            )
+            if (
+                OrganizationManagementLevel(user.get("organization_management_level"))
+                > permstore.user_oml
+            ):
+                raise PermissionDenied(
+                    f"Your organization management level is not high enough to reset a Level of {user.get('organization_management_level')}!"
```
Comment (jsangmeister) [inline openslides_backend/action/actions/user/create_update_permissions_mixin.py:372]: error message does not match the error

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py [hunk 4]
```diff
@@ -196,11 +196,12 @@ def check_permissions(self, instance: Dict[str, Any]) -> None:
         if permstore.user_oml == OrganizationManagementLevel.SUPERADMIN:
             return
 
+        self._check_for_higher_OML(permstore, instance)
         self._check_OML_in_instance(permstore, instance)
         actual_group_fields = self._get_actual_grouping_from_instance(instance)
 
         # Ordered by supposed velocity advantages. Changing order only can effect the sequence of detected errors for tests
-        self.check_group_E(permstore, actual_group_fields["E"])
+        self.check_group_E(permstore, actual_group_fields["E"], instance)
```
Comment (jsangmeister) [inline openslides_backend/action/actions/user/create_update_permissions_mixin.py:204]: Please revert the change to the params since it is no longer needed

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Also add a test which tries to add a basic property from group A, but fails because the OML is higher than the one from the request user.

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/action/actions/user/create_update_permissions_mixin.py:372]: Update to f`"Your organization management level is not high enough to change a user with a Level of {user.get('organization_management_level')}!"`

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py (hunk 4 above)
Comment (reiterl) [reply openslides_backend/action/actions/user/create_update_permissions_mixin.py:204]: Done.

File: openslides_backend/action/actions/user/create_update_permissions_mixin.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/actions/user/create_update_permissions_mixin.py:362]: No, I need this check, see the rest of the code. There it is always checked and the test_create.py failed, if it is missing.

Comment (jsangmeister) [reply openslides_backend/action/actions/user/create_update_permissions_mixin.py:360]: ok

**Review by jsangmeister — APPROVED** [review]
